core/modules/book/book.module

```diff
@@ -1294,3 +1307,72 @@ function book_menu_subtree_data($link) {
 
   return $tree[$cid];
 }
+
+/**
+ * Invalidates cached data relating to book_cache.
+ *
+ * @param int|null $cid
+ *   (optional) Cache ID of the record to clear. If empty, all records will be
+ *   cleared from the table. Defaults to NULL.
+ * @param bool $wildcard
+ *   (optional) If TRUE, cache IDs starting with $cid are deleted in addition to
+ *   the exact cache ID specified by $cid. Defaults to FALSE.
+ */
+function _book_cache_clear($cid = NULL, $wildcard = FALSE) {
+  cache_clear_all($cid, 'cache_book', $wildcard);
+}
+
+/**
+ * Implements hook_flush_caches().
+ */
+function book_flush_caches() {
+  cache_flush('cache_book');
```

Comment:
As pointed out by @avpaderno, this hook implementation should clear caches by itself. Rather, it should return an array of cache bins that will be cleared by the invoking function.